langchain4j-open-ai/src/main/java/dev/langchain4j/model/openai/OpenAiEmbeddingModel.java

```diff
@@ -102,7 +104,39 @@ public Response<List<Embedding>> embedAll(List<TextSegment> textSegments) {
                 .map(TextSegment::text)
                 .toList();
 
-        return embedTexts(texts);
+        List<List<String>> textBatches = splitList(texts, REQUEST_TEXT_MAX_ARRAY_SIZE);
+
+        return embedBatchedTexts(textBatches);
+    }
+
+    private List<List<String>> splitList(final List<String> inputList, final int size) {
+        if (size <= 0) {
+            throw new IllegalArgumentException("Size must be greater than 0");
```

Comment:
If this value will be configurable, it would be nice to do this validation in the constructor, to fail fast:

`ensureGreaterThanZero(maxSegmentsPerBatch, "maxSegmentsPerBatch")`